jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/BaseGenerator.java

```diff
@@ -289,4 +294,36 @@ protected void addSchemaLabels(BuildConfiguration.BuildConfigurationBuilder buil
             buildBuilder.labels(labels);
         }
     }
-}
+
+    public void checkAndWarnIfProjectHasNotBeenBuilt() {
+      File finalOutputArtifact = null;
+      try {
+        finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(getProject());
+        if (finalOutputArtifact == null) {
+          log.error(
+              "Final output artifact file was not detected. The project may have not been built. " +
+                  "HINT: try to compile and package your application prior to running the container image build task.");
+          return;
+        }
+        long currentLastModifiedTime = finalOutputArtifact.lastModified();
+        long previousLastModifiedTime = retrievePreviousArtifactLastModifiedTime(getProject().getBuildDirectory().toPath());
+        boolean isCurrentFinalArtifactSameAsPrevious = currentLastModifiedTime == previousLastModifiedTime;
+
+        if (isCurrentFinalArtifactSameAsPrevious) {
+          log.info(
+              "Final output artifact is the same as previous build. " +
```

Comment:
I am wondering if it is worth making it in 2 lines (HINT: ... in a new line)

Reply:
no I mean in the output when you run the goal: showing to the user in 2 lines instead of 1